CMakeLists.txt

```diff
@@ -384,14 +327,147 @@ function( target_link_exe_with_core target )
+    if( NOT Python3_FOUND )
+        find_package( Python3 REQUIRED)
+    endif()
+
+    # Macro for adding a new GoogleTest test
+    function(add_gtest testName ENGINE debug testSource )
+	    if ("{$ENGINE}" STREQUAL "")
+		    message(FATAL_ERROR "engine cannot be empty, ever!")
+	    endif()
+	    add_executable(${testName} ${testSource} ${ARGN})
+	    target_compile_definitions(${testName} PUBLIC LPF_CORE_IMPL_ID=${ENGINE})
+	    target_compile_definitions(${testName} PUBLIC LPF_CORE_MPI_USES_${ENGINE})
+	    if (debug)
+		    target_include_directories( ${testName} BEFORE PRIVATE ${CMAKE_SOURCE_DIR}/include/debug )
+		    target_link_libraries(${testName} lpf_debug lpf_hl_debug GTest::gtest GTest::gtest_main)
+	    else(debug)
+		    target_link_libraries(${testName} GTest::gtest GTest::gtest_main)
+	    endif(debug)
+
+
+	    # Extract test-specific information from comments of tests
+	    file(READ ${testSource} fileContents)
+	    string(REGEX MATCH "Exit code: ([0-9]+)" _ ${fileContents})
+	    set(retCode ${CMAKE_MATCH_1})
+	    string(REGEX MATCH "pre P >= ([0-9]+)"  _ ${fileContents})
+	    set(minProcs ${CMAKE_MATCH_1})
+	    string(REGEX MATCH "pre P <= ([0-9]+)" _ ${fileContents})
+	    set(maxProcs ${CMAKE_MATCH_1})
+	    string(REGEX MATCH "-probe ([0-9]+.[0-9]+)" _ ${fileContents})
+	    set(lpfProbeSecs ${CMAKE_MATCH_1})
+
+	    # The "\pre P <= max" comment in a test indicates the desired number of
+	    # maximum LPF processes. If, at CMake configuration, the number of detected
+	    # processes is smaller, it overrides the desired number of maximum LPF
+	    # processes. If the test does not define a desired number of maximum LPF
+	    # processes, it will be set to the detected number of processes.
+	    #
+	    # The "\pre P >= min" comment in a test indicates the desired number of
+	    # minimum LPF processes. If the number of detected processes is smaller, it
+	    # overrides the desired number of minimum LPF processes. If the test does
+	    # not define a desired minimum number of LPF processes, it will be set to
+	    # the detected number of processes.
+	    #
+	    # Hence if neither a desired min nor max number of processes are defined in
+	    # a test, then the test will execute using the number of detected
+	    # processes only.
+	    #
+	    # Most tests only define a mininum number of desired processes, such as
+	    # "\pre P >= 1". In those cases, the test will execute for all possible
+	    # number of processes between 1 and the detected number of processes. This
+	    # has been the behaviour of the test suite since v1.0.
+
+	    target_link_exe_with_core(${testName} ${ENGINE})
+
+
+	    ProcessorCount(nprocs)
+	    if ("${minProcs}" STREQUAL "")
+		    ProcessorCount(minProcs)
+	    endif()
+	    if ("${maxProcs}" STREQUAL "")
+		    ProcessorCount(maxProcs)
```

Comment:
There should also be a constant maximum enforced here (like 4 or 5). It is probably a good idea to make this constant maximum a CMake cache variable.

Reply:
Should be fixed.